File: packages/agent/src/events/types/context.ts
/**
 * @fileoverview Context Events
 *
 * Events for context clearing operations.
 */

import type { BaseEvent } from './base.js';

// =============================================================================
// Context Clearing Events
// =============================================================================

/**
 * Context cleared event - marks where context was cleared
 * Unlike compaction, no summary is preserved - all messages before this point are discarded
 */
export interface ContextClearedEvent extends BaseEvent {
  type: 'context.cleared';
  payload: {
    /** Token count before clearing */
    tokensBefore: number;
    /** Token count after clearing (system prompt + tools only) */
    tokensAfter: number;
    /** Reason for clearing */
    reason: 'manual';
  };
}

File: packages/agent-rs/crates/tron-events/src/types/payloads/context.rs
//! Context event payloads.

use serde::{Deserialize, Serialize};

/// Payload for `context.cleared` events.
#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ContextClearedPayload {
    /// Token count before clearing.
    pub tokens_before: i64,
    /// Token count after clearing.
    pub tokens_after: i64,
    /// Reason for clearing.
    pub reason: String,
}
